docs/guide.rst

```diff
@@ -1,108 +1,210 @@
+.. jupyter-execute::
+   :hide-code:
 
+   circuit.draw("mpl")
 
-To retrieve wait for a result to be available and retrieve it, use
+to the transpiled one:
 
 .. jupyter-execute::
 
-    result = job.result()
+   transpiled_circuit = qiskit.transpile(circuit, backend, optimization_level=2)
+   transpiled_circuit.draw("mpl")
+
+
+Transpiler bypass
+-----------------
+
+.. warning:: We highly recommend to always use the built-in transpiler, at least with ``optimization_level=0``. This guarantees that the quantum circuit is valid for submission to the AQT cloud. In particular, it wraps the gate parameters to fit in the restricted ranges accepted by the `AQT API <https://arnica-stage.aqt.eu/api/v1/docs>`_. In addition, higher optimization levels may significantly improve the circuit execution speed.
 
-The Qiskit `Result` object contains the outcome of all circuit samples:
+If a circuit is already defined in terms of the :ref:`native gates set <basis-gates>` with their restricted parameter ranges and no optimization is wanted, it can be submitted for execution without any additional transformation using the :meth:`AQTResource.run <qiskit_aqt_provider.aqt_resource.AQTResource.run>` method:
 
 .. jupyter-execute::
 
-    print(result.get_counts())
+   native_circuit = qiskit.QuantumCircuit(2)
+   native_circuit.rxx(pi/2, 0, 1)
+   native_circuit.r(pi, 0, 0)
+   native_circuit.r(pi, pi, 1)
+   native_circuit.measure_all()
+
+   job = backend.run(native_circuit)
+   result = job.result()
+
+   if result.success:
+       print(result.get_counts())
+   else:
+       raise RuntimeError
+
+Circuits that do not satisfy the AQT API restrictions are rejected by raising a :class:`ValueError` exception.
+
+Transpiler plugin
+-----------------
+
+The built-in transpiler largely leverages the :mod:`qiskit.transpiler`. Custom passes are registered in addition to the presets, irrespective of the optimization level, to ensure that the transpiled circuit is compatible with the restricted parameter ranges accepted by the `AQT API <https://arnica-stage.aqt.eu/api/v1/docs>`_:
+
+* in the translation stage, the :class:`WrapRxxAngles <qiskit_aqt_provider.transpiler_plugin.WrapRxxAngles>` pass exploits the periodicity of the :class:`RXXGate <qiskit.circuit.library.RXXGate>` to wrap its angle θ to the [0, π/2] range. This may come at the expense of extra single-qubit rotations.
+* in the scheduling stage, the :class:`RewriteRxAsR <qiskit_aqt_provider.transpiler_plugin.RewriteRxAsR>` pass rewrites :class:`RXGate <qiskit.circuit.library.RXGate>` operations as :class:`RGate <qiskit.circuit.library.RGate>` and wraps the angles θ∈[0, π] and φ∈[0, 2π]. This does not restrict the generality of quantum circuits and enables efficient native implementations.
+
+.. warning:: Circuits accepted by the AQT API are executed after applying one further transformation. Small-angle θ instances of :class:`RGate <qiskit.circuit.library.RGate>` are substituted as
+
+  R(θ, φ) → R(π, π)·R(θ+π, φ)
```

Comment:
would it be worth using LaTeX here (and above)? I see that mathjax would be available

Reply:
Done. But only in the user guide. I find it more readable to keep the unicode math for the docstrings.